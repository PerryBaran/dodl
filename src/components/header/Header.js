import React, { useContext } from 'react';
import AppContext from '../AppContext';
import style from './header.module.css';

const Header = (props) => {
    const { hideWhilePlaying } = useContext(AppContext);

    return (
        <header className={`centerFlex ${style.headerContainer}`}>
          <h1 className={`${style.header} ${hideWhilePlaying(style.pause)}`}>Dreaming of Detuned Love</h1>
        </header>
    );
}

export default Header;