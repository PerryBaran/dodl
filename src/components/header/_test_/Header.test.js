import React from "react";
import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom'
import Header from "../Header";
import AppContext from "../../AppContext";
import renderer from "react-test-renderer";

afterEach(cleanup);

const RenderWithContext = (isPlaying = false) => {
    const hideWhilePlaying = (input) => {
        if (!isPlaying) {
          return input
        }
        return ''
    };
    return (
        <AppContext.Provider value={{hideWhilePlaying}}>
            <Header/>
        </AppContext.Provider>
    )
};

it("renders without crashing", () => { 
    render(RenderWithContext());
});

it("shows className while paused", () => {
    render(RenderWithContext());
    const h1Element = screen.getByText('Dreaming of Detuned Love')
    expect(h1Element).toHaveClass('pause');
});

it("hides className while playing", () => {
    render(RenderWithContext(true));
    const h1Element = screen.getByText('Dreaming of Detuned Love')
    expect(h1Element).not.toHaveClass('pause');
});

it("matches snapshot while paused", () => {
    const tree = renderer.create(RenderWithContext()).toJSON();
    expect(tree).toMatchSnapshot();
});

it("matches snapshot while playing", () => {
    const tree = renderer.create(RenderWithContext(true)).toJSON();
    expect(tree).toMatchSnapshot();
});
