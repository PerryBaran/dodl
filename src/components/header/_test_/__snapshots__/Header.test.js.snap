// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot while paused 1`] = `
<header
  className="centerFlex headerContainer"
>
  <h1
    className="header pause"
  >
    Dreaming of Detuned Love
  </h1>
</header>
`;

exports[`matches snapshot while playing 1`] = `
<header
  className="centerFlex headerContainer"
>
  <h1
    className="header "
  >
    Dreaming of Detuned Love
  </h1>
</header>
`;
